libp2p/protocols/pubsub/pubsub.nim

```diff
@@ -85,7 +85,7 @@ type
   TopicPair* = tuple[topic: string, handler: TopicHandler]
 
   MsgIdProvider* =
-    proc(m: Message): MessageID {.noSideEffect, raises: [Defect], gcsafe.}
+    proc(m: Message): Result[MessageID, string] {.noSideEffect, raises: [Defect], gcsafe.}
```

Comment:
so per discussion elsewhere, this should return a `ValidationResult` so that gossip can penalize peers that send junk

Reply:
Updated.